container/statistics.go

```diff
@@ -0,0 +1,309 @@
+// +build linux
+
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import "fmt"
+
+type GetStatOpt map[string]interface{}
+
+func (opt GetStatOpt) GetStringValue(key string) (string, error) {
+	val, exists := opt[key]
+	if exists {
+		return val.(string), nil
```

Comment:
Consider adding checking result of type assertion. It might look like below:
```
func (opt GetStatOpt)getStringValue(key string)(string, error) {
if val, exists := opt[key]; exists {
	if valStr, ok := val.(string); ok {
		return valStr, nil
	} else {
	    return "", fmt.Errorf("Value for key `%s` is expected to be string, but it isn't", key)
	}
}	
	return "", fmt.Errorf("Cannot find value for key `%s`", key)
}
```